packages/epics/src/agreements/components/agreement-card.tsx

```diff
@@ -0,0 +1,125 @@
+import { Text } from '@radix-ui/themes';
+import { Card, Badge, Skeleton } from '@hypha-platform/ui';
+import Image from 'next/image';
+import { EyeOpenIcon, ChatBubbleIcon } from '@radix-ui/react-icons';
+
+type Creator = {
+  avatar?: string;
+  name?: string;
+  surname?: string;
+};
+
+type AgreementCardProps = {
+  creator?: Creator;
+  title?: string;
+  commitment?: number;
+  status?: string;
+  views?: number;
+  comments?: number;
+  isLoading?: boolean | undefined;
+};
+
+const statusBadge = (status: string, isLoading: boolean | undefined) => {
+  switch (status) {
+    case 'active':
+      return (
+        <Badge isLoading={isLoading} variant="positive">
+          Active
+        </Badge>
+      );
+    case 'voting':
+      return (
+        <Badge isLoading={isLoading} variant="warning">
+          On voting
+        </Badge>
+      );
+    case 'completed':
+      return (
+        <Badge isLoading={isLoading} variant="action">
+          Completed
+        </Badge>
+      );
+    case 'rejected':
+      return (
+        <Badge isLoading={isLoading} variant="destructive">
+          Rejected
+        </Badge>
+      );
+  }
+};
```

Comment:
I'd refactor this like so and put to a seperate file `status-badge.tsx`:

```suggestion
const BADGE_VARIANTS: Record<string, BadgeProps> = {
  active: { variant: 'positive', children: 'Active' },
  voting: { variant: 'warning', children: 'On voting' },
  completed: { variant: 'action', children: 'Completed' },
  rejected: { variant: 'destructive', children: 'Rejected' },
};

const StatusBadge = ({
  status,
  isLoading,
}: {
  status?: string;
  isLoading: boolean | undefined;
}) => {
  const badge = BADGE_VARIANTS[status as keyof typeof BADGE_VARIANTS];

  return status ? <Badge isLoading={isLoading} {...badge} /> : null;
};
```